File: 7KYU/GCD sum/solution.clj
(ns gcd-sum.core)

(defn gcd [a b]
  (if (zero? b)
    a
    (recur b (mod a b))))

(defn solve [s g]
  (let [y (- s g)]
    (if (= g (gcd g y))
      [g y]
      -1)))

File: 7KYU/GCD sum/solution.go
package kata

func gcd(a, b int) int {
	for {
		if b == 0 {
			return a
		}
		c := a % b
		a = b
		b = c
	}
}

func Solve(s int, g int) []int {
	y := s - g
	if g == gcd(g, y) {
		return []int{g, y}
	} else {
		return []int{-1, -1}
	}
}
